internal/pkg/pkg.go

```diff
@@ -169,6 +171,26 @@ func (p *Pkg) DirectSubpackages() ([]*Pkg, error) {
 	return subPkgs, nil
 }
 
+// adjustDisplayPathForSubpkg adjusts the display path of subPkg relative to the RootPkgUniquePath
+// subPkg also inherits the RootPkgUniquePath value from parent package p
+func (p *Pkg) adjustDisplayPathForSubpkg(subPkg *Pkg) error {
+	if string(p.RootPkgUniquePath) == "" {
+		// this means p is the root package on which command is originally invoked
+		subPkg.RootPkgUniquePath = p.UniquePath
+	} else {
+		// inherit the RootPkgUniquePath from the parent package
+		subPkg.RootPkgUniquePath = p.RootPkgUniquePath
+	}
+	// display path of subPkg should be relative to parent dir of rootPkg
+	// e.g. if mysql(subPkg) is direct subpackage of wordpress(p), DisplayPath of "mysql" should be "wordpress/mysql"
+	dp, err := filepath.Rel(filepath.Dir(string(subPkg.RootPkgUniquePath)), string(subPkg.UniquePath))
```

Comment:
https://github.com/GoogleContainerTools/kpt/blob/next/internal/types/types.go#L53

If I had a dollar for every time someone assumed POSIX filesystem, I'd be a rich man :)

As it stands, DisplayPath is a slash-delimited os-agnostic path, but you're using filepath package. So either:

We're changing the semantics of DisplayPath (which there's no indication it's happening in this PR)
You should adhere to that semantic.

The intent of having different types was to eliminate that conflation.

Reply:
How does using `filepath` package make it non os-agnostic path, we are using the `filepath` package elsewhere as well.

I changed the semantic of DisplayPath and updated description.

Reply:
Because filepath package works on OS paths: https://golang.org/pkg/path/filepath/

If you want to convert to/from slash-separated path, you need to do that explicitly.